src/protocol/attribution/accumulate_credit.rs

```diff
@@ -268,34 +272,32 @@ mod tests {
             [1, 1, 0, 1],
             [0, 0, 1, 0],
             [1, 0, 0, 10],
+            [0, 0, 2, 0],
+            [1, 1, 0, 3],
+            [1, 1, 0, 12],
+            [0, 1, 2, 0],
+            [0, 1, 2, 0],
+            [1, 1, 0, 6],
+            [1, 1, 0, 4],
+            [0, 1, 5, 0],
+            [1, 1, 5, 6],
+        ];
+        const EXPECTED: &[u128] = &[
+            0, 0, 19, 19, 9, 7, 6, 1, 0, 10, 15, 15, 12, 0, 10, 10, 4, 6, 6,
```

Comment:
Adding more events to catch some corner case bugs. This is quite complicated protocol/code so we need a better set of test cases.